neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/bolt/BoltNeo4jResultSet.java

```diff
@@ -143,13 +151,19 @@ private void flattenNode(Node node, String nodeKey) {
 			classes.add(InternalTypeSystem.TYPE_SYSTEM.LIST());
 		}
 		for (String key : node.keys()) {
-			if (keys.indexOf(nodeKey + "." + key) == -1) {
-				keys.add(nodeKey + "." + key);
+			String newKey = parsePropertyKey(nodeKey, key);
+			if (keys.indexOf(newKey) == -1) {
+				keys.add(newKey);
 				classes.add(node.get(key).type());
 			}
 		}
 	}
 
+	private String parsePropertyKey(String nodeKey, String key) {
+		String keyTemplate = SPECIAL_CHARS.matcher(key).find() ? FLATTEN_KEY_TEMPLATE_BACKTICK : FLATTEN_KEY_TEMPLATE;
```

Comment:
shouldn't this be part of a `quote` method?

I think that would make it easier, like I do it in other places:

```
def quote (name:String) = if (name.isSpecial()) '`'+sanitize(name)+'`' else name
```

and use that in all places instead all the string formats?

Reply:
same for labels and rel-types

Reply:
isSpecial could also use the 

Character.isJavaIdentifierStart(c) 
Character.isJavaIdentifier(c)

instaed of regexp